projects/astc-encoder/astc_load_decompress_image_fuzzer.cc

```diff
@@ -0,0 +1,127 @@
+
+  if (size < 20)
+    return 0;
+
+  astc_compressed_image image_comp;
+  astcenc_error error;
+  astcenc_profile profile;
+  int out_bitness;
+  FuzzedDataProvider stream(data, size);
+
+  const astcenc_profile profiles[] = {ASTCENC_PRF_LDR, ASTCENC_PRF_LDR_SRGB,
+                                      ASTCENC_PRF_HDR,
+                                      ASTCENC_PRF_HDR_RGB_LDR_A};
+
+  int profile_type = stream.ConsumeIntegralInRange<int>(0, 3);
+  profile = profiles[profile_type];
+  out_bitness = 8 << (profile_type / 2);
+
+  // Avoid dividing by zero
+  uint8_t block_x = stream.ConsumeIntegral<uint8_t>();
+  if (!block_x)
+    block_x++;
+
+  uint8_t block_y = stream.ConsumeIntegral<uint8_t>();
+  if (!block_y)
+    block_y++;
+
+  uint8_t block_z = stream.ConsumeIntegral<uint8_t>();
+  if (!block_z)
+    block_z++;
+
+  std::vector<uint8_t> dim_x_vector = stream.ConsumeBytes<uint8_t>(3);
+  std::vector<uint8_t> dim_y_vector = stream.ConsumeBytes<uint8_t>(3);
+  std::vector<uint8_t> dim_z_vector = stream.ConsumeBytes<uint8_t>(3);
+  uint8_t *dim_x_data = dim_x_vector.data();
+  uint8_t *dim_y_data = dim_y_vector.data();
+  uint8_t *dim_z_data = dim_z_vector.data();
+
+  // Dimensions cannot be zero
+  unsigned int dim_x =
+      unpack_bytes(*dim_x_data, *(dim_x_data + 1), *(dim_x_data + 2));
+  if (!dim_x)
+    dim_x++;
+
+  unsigned int dim_y =
+      unpack_bytes(*dim_y_data, *(dim_y_data + 1), *(dim_y_data + 2));
+  if (!dim_x)
+    dim_x++;
+
+  unsigned int dim_z =
+      unpack_bytes(*dim_z_data, *(dim_z_data + 1), *(dim_z_data + 2));
+  if (!dim_x)
+    dim_x++;
+
+  unsigned int xblocks = (dim_x + block_x - 1) / block_x;
+  unsigned int yblocks = (dim_y + block_y - 1) / block_y;
+  unsigned int zblocks = (dim_z + block_z - 1) / block_z;
+  unsigned int buffer_size = xblocks * yblocks * zblocks * 16;
+  if (size - 20 < buffer_size)
+    return 0;
+
+  std::vector<uint8_t> buffer = stream.ConsumeBytes<uint8_t>(buffer_size);
+
+  image_comp.data = buffer.data();
+  image_comp.data_len = buffer.size();
+  image_comp.block_x = block_x;
+  image_comp.block_y = block_y;
+  image_comp.block_z = block_z;
+  image_comp.dim_x = dim_x;
+  image_comp.dim_y = dim_y;
+  image_comp.dim_z = dim_z;
+
+  astcenc_config config{};
+  astcenc_preset preset = ASTCENC_PRE_FAST;
+  error = astcenc_config_init(profile, image_comp.block_x, image_comp.block_y,
+                              image_comp.block_z, preset, 0, config);
+  if (error)
+    return 0;
+
+  // Initialize cli_config_options with default values
+  cli_config_options cli_config{
+      0,
+      1,
+      false,
+      false,
+      -10,
+      10,
+      {ASTCENC_SWZ_R, ASTCENC_SWZ_G, ASTCENC_SWZ_B, ASTCENC_SWZ_A},
+      {ASTCENC_SWZ_R, ASTCENC_SWZ_G, ASTCENC_SWZ_B, ASTCENC_SWZ_A}};
+
+  astcenc_context *codec_context;
+  error =
+      astcenc_context_alloc(config, cli_config.thread_count, &codec_context);
+  if (error)
```

Comment:
Optional: you can combine these lines like

```c++
if (astcenc_context_alloc(...) != 0)
  return 0;
```

Reply:
Aha, https://github.com/ARM-software/astc-encoder/blob/master/Source/astcenc.h#L172 defines `ASTCENC_SUCCESS` and error codes, so you can write

```c++
if (astcenc_context_alloc(...) != ASTCENC_SUCCESS)
  return 0;
```

which is even clearer.

Reply:
And please do the same for `astcenc_config_init` above.